odl/tomo/geometry/spect_geometry.py

```diff
@@ -0,0 +1,129 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""SPECT geometry."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+# External
+import numpy as np
+
+# Internal
+import odl
+from odl.tomo.geometry.detector import Flat2dDetector
+from odl.tomo.geometry.geometry import (Geometry,
+                                        AxisOrientedGeometry)
+from odl.tomo.util.utility import perpendicular_vector
+
+__all__ = ('ParallelHoleCollimatorGeometry', )
+
+
+class ParallelHoleCollimatorGeometry(Geometry, AxisOrientedGeometry):
+    """Geometry for SPECT Parallel hole collimator."""
+
+    def __init__(self, apart, dpart, det_radius, axis=[0, 0, 1],
+                 **kwargs):
+        """Initialize a new instance.
+        Parameters
+        ----------
+        apart : 1-dim. `RectPartition`
+            Partition of the angle interval
+        dpart : 2-dim. `RectPartition`
+            Partition of the detector parameter rectangle
+        det_radius : `list` of positive `float`
+            A list of radii for every position of the detector
+            Individual radii must be positive.
+            If all radii are equal, `list` of one radius is sufficient,
+            otherwise a radius per angular position is needed.
+        axis : `array-like`, shape ``(3,)``, optional
+            Fixed rotation axis.
+        fwhm : `float`, optional
+            Full width at half maximum
+        center_to_det : `array-like`, shape ``(2,)``, optional
+            Initial state of the vector pointing from the center of
+            rotation to the detector reference point.
+            The zero vector is not allowed.
+            By default, a `perpendicular_vector` to ``axis`` is used.
+
+        """
+        AxisOrientedGeometry.__init__(self, axis)
+
+        bool_list = []
+        for ii in range(len(det_radius)):
+            bool_list.append((odl.util.all_equal(det_radius[0],
+                                                 det_radius[ii])))
+        if False not in bool_list:
+            self._equal_radii = True
+        else:
+            if len(det_radius) != len(apart.grid.points()):
+                raise ValueError('detector radii has to be defined for all '
+                                 'angular positions, got {} angular positions.'
+                                 ''.format(len(apart.grid.points())))
```

Comment:
Some clean-up needed. If you use the code I posted above, you have the list of radii. Then you can simply store
```python
self._all_radii_equal = all(r == rad_list[0] for r in rad_list)
```